WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2316,9 +2320,51 @@ private GutenbergPropsBuilder getGutenbergPropsBuilder() {
         );
     }
 
-    private Boolean canViewEditorOnboarding() {
-        return (mAccountStore.getAccount().getUserId() % 100 >= (100 - EDITOR_ONBOARDING_PHASE_PERCENTAGE)
-                || BuildConfig.DEBUG);
+    /**
+     * Temporary method for the Editor Onboarding project to control the percentage
+     * of users able to use this new editor onboarding tooltip feature. This will eventually
+     * be removed when the functionality is opened to all users.
+     */
+    private boolean canViewEditorOnboarding() {
+        // Get the userId for the logged in user. If not WPcom, use the anonymous userId
+        // generated by tracks.
+        long userAccountId = mAccountStore.getAccount().getUserId();
+        String userId = String.valueOf(userAccountId);
+        if (userAccountId == 0) {
+            SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
+            userId = preferences.getString("nosara_tracks_anon_id", null);
```

Comment:
Instead of hardcoding this string, would we be better off reusing it from `AnalyticsTrackerNosaroa`? We would need to expose either the `TRACKS_ANON_ID` constant or the `getAnonIdPrefKey`, but that seems preferable to repeating it here. Or, perhaps we should be using `Tracker::getAnonId` here, instead?

Reply:
Thanks for the review @mchowning! I thought the same thing as well, but since this hashing mechanism is a temporary solution and will only be in place until the editor onboarding project is rolled out to all users, I didn't want to make a bunch of changes in the more permanent supporting code. I'm good leaving this string as-is for this implementation. 👍